dydx3/starkex/transfer.py

```diff
@@ -0,0 +1,118 @@
+from collections import namedtuple
+import math
+
+from dydx3.constants import COLLATERAL_ASSET
+from dydx3.constants import COLLATERAL_ASSET_ID_BY_NETWORK_ID
+from dydx3.starkex.constants import ONE_HOUR_IN_SECONDS
+from dydx3.starkex.constants import TRANSFER_FIELD_BIT_LENGTHS
+from dydx3.starkex.constants import TRANSFER_PADDING_BITS
+from dydx3.starkex.constants import TRANSFER_PREFIX
+from dydx3.starkex.constants import TRANSFER_FEE_ASSET_ID
+from dydx3.starkex.constants import TRANSFER_MAX_AMOUNT_FEE
+from dydx3.starkex.helpers import nonce_from_client_id
+from dydx3.starkex.helpers import to_quantums_exact
+from dydx3.starkex.signable import Signable
+from dydx3.starkex.starkex_resources.proxy import get_hash
+
+StarkwareTransfer = namedtuple(
+    'StarkwareTransfer',
+    [
+        'sender_position_id',
+        'receiver_position_id',
+        'receiver_public_key',
+        'quantums_amount',
+        'nounce',
+        'expiration_epoch_hours',
+    ],
+)
+
+
+class SignableTransfer(Signable):
+    """
+    Wrapper object to convert a transfer, and hash, sign, and verify its
+    signature.
+
+    Adapted from dydxprotocol/starkex-lib/src/signable/transfer.ts
+    """
+
+    def __init__(
+        self,
+        sender_position_id,
+        receiver_position_id,
+        receiver_public_key,
+        human_amount,
+        client_id,
+        # adapted from expiration_iso_timestamp in transfer.ts, to be
+        # consistent with params of SignableConditionalTransfer.
+        expiration_epoch_seconds,
+        network_id,
+    ):
+        nounce = nonce_from_client_id(client_id)
+
+        # The transfer asset is always the collateral asset.
+        quantums_amount = to_quantums_exact(
+            human_amount, COLLATERAL_ASSET)
```

Comment:
each arg should go on its own line with the parenthesis on a final line

Reply:
Done.